### PR TITLE
Fix race condition on reminder unregister

### PR DESCRIPTION
**Pull request by artursouza**
Signed-off-by: Artur Souza <artursouza.ms@outlook.com>

# Description

Fix race condition on reminder unregister.

## Issue reference

Please reference the issue this PR will close: N/A

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Comment by artursouza** [comment]
ok-to-test

**Review by ItalyPaleAle — DISMISSED** [review]

File: pkg/actors/actors.go [hunk 1]
```diff
@@ -1841,6 +1847,7 @@ func (a *actorsRuntime) DeleteReminder(ctx context.Context, req *DeleteReminderR
 		return err
 	}
 
+	log.Infof("Deleted reminder with key: %v", reminderKey)
```
[the remark this answers is not in the document]
Comment (yaron2) [reply pkg/actors/actors.go:1850]: Should be Debug.

[the remark this answers is not in the document]
Comment (artursouza) [reply pkg/actors/actors.go:1850]: I forgot to remove this. Sorry. This was for my local debugging only.

**Review by ItalyPaleAle — APPROVED** [review]

**Comment by artursouza** [comment]
This change is not enough since the E2E test failed in another instance here: https://github.com/dapr/dapr/actions/runs/3441358778/jobs/5741116555

**Review by ItalyPaleAle — APPROVED** [review]

**Comment by artursouza** [comment]
I had added resiliency for the E2E test apps, and @ItalyPaleAle is fixing race conditions over there. So, I will make this fix without resiliency first. If it proves stable in ok-to-test, I will cut a hotfix in runtime.

**Pull request merged**